ouroboros-network/src/Ouroboros/Network/Server/ConnectionTable.hs

```diff
@@ -0,0 +1,214 @@
+
+{-# LANGUAGE BangPatterns        #-}
+{-# LANGUAGE FlexibleContexts    #-}
+{-# LANGUAGE RankNTypes          #-}
+{-# LANGUAGE RecordWildCards     #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Ouroboros.Network.Server.ConnectionTable (
+      ConnectionTable
+    , ConnectionTableRef (..)
+    , ValencyCounter
+
+    , newConnectionTable
+    , refConnection
+    , addConnection
+    , removeConnection
+    , newValencyCounter
+    , addValencyCounter
+    , remValencyCounter
+    , waitValencyCounter
+    , readValencyCounter
+    ) where
+
+import           Control.Monad (when)
+import           Control.Monad.Class.MonadSTM
+--import           Control.Tracer XXX Not Yet
+import qualified Data.Map.Strict as M
+import           Data.Set (Set)
+import qualified Data.Set as S
+import qualified Network.Socket as Socket
+import           Text.Printf
+
+-- A ConnectionTable represent a set of connections that is shared between
+-- servers and subscription workers. Its main purpose is to avoid the creation of duplicate
```

Comment:
A typo: it should be `It's main purpose`

Reply:
Will fix.